### PR TITLE
Date selector component

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR adds the date/time selector component that is used to set the trip departure/arrival times.

### CODE REVIEW

File: packages/trip-form/src/DateTimeSelector/index.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+import {
+  OTP_API_DATE_FORMAT,
+  OTP_API_TIME_FORMAT
+} from "@opentripplanner/core-utils/lib/time";
+
+import ModeButton from "../ModeButton";
+import { DepartureRow, DateTimeRow } from "./styled";
+
+function checkInput(type) {
```
Comment (evansiroky) [inline packages/trip-form/src/DateTimeSelector/index.js]: Can you please create some jsDoc for this function?

**Comment by binh-dam-ibigroup** [comment]
@fpurcell That is a good question... GitHub is trying to make me use these dependencies that don't seem to work... I'll revert both to to 0.0.9 for now...
`
    "@opentripplanner/core-utils": "^0.0.10",
    "@opentripplanner/icons": "^0.0.10"
`


> I am yet to get `yarn dev` to work correctly with this PR, @binh-dam-ibigroup . Any ideas?
> 
> Module parse failed: Unexpected token (25:9)
> File was processed with these loaders:
> 
> * ./node_modules/@storybook/addon-storysource/loader.js
>   You may need an additional loader to handle the result of these loaders.
> 
> ./packages/trip-form/node_modules/@opentripplanner/icons/src/index.story.js@http://localhost:5555/vendors~main.c9e83e96cb48e4d78b6d.bundle.js:263656:7

**Comment by binh-dam-ibigroup** [comment]
@evansiroky are these two versions of these packages published to master? 
"@opentripplanner/core-utils": "^0.0.10", "@opentripplanner/icons": "^0.0.10"
This branch is in sync with master right now but @fpurcell and I are not able to build using these.

**Comment by fpurcell** [comment]
> @evansiroky are these two versions of these packages published to master?
> "@opentripplanner/core-utils": "^0.0.10", "@opentripplanner/icons": "^0.0.10"

These versions are out there on master:  https://www.npmjs.com/org/opentripplanner
And afaik, they do build and run.  I rebuilt Storybook successfully when I released last week.
Thinking there was another storybook plugin added recently.  FWIW, I rebuilding master again
just to check...

**Comment by evansiroky** [comment]
Yeah, I encountered these errors as well. I resolved them by making sure each`@opentripplanner/*` package was referencing the version that existed in the branch we were working on. You will want to look in each package's package.json to make sure the correct dependencies exist and also make sure that there aren't any `node_modules/@opentripplanner/*` installations in each subpackage. I'd merge `@opentripplanner/master` into your branch and the correct things as needed from there.

**Comment by fpurcell** [comment]
BTW @binh-dam-ibigroup ... github complaining about packages/trip-form/package.json preventing merge.

**Pull request merged**